common/src/main/java/com/wynntils/mc/mixin/AbstractContainerScreenMixin.java

```diff
@@ -134,26 +133,22 @@ private void mousePressedPre(double mouseX, double mouseY, int button, CallbackI
     }
 
     @Inject(method = "mouseDragged(DDIDD)Z", at = @At("RETURN"))
-    private void mouseDraggedPre(
+    private void mouseDraggedPost(
             double mouseX,
             double mouseY,
             int button,
             double deltaX,
             double deltaY,
             CallbackInfoReturnable<Boolean> cir) {
-        TextInputBoxWidget focusedTextInput = ((TextboxScreen) this).getFocusedTextInput();
-
-        if (focusedTextInput != null) {
-            focusedTextInput.mouseDragged(mouseX, mouseY, button, deltaX, deltaY);
+        for (GuiEventListener listener : ((AbstractContainerScreen<?>) (Object) this).children()) {
+            listener.mouseDragged(mouseX, mouseY, button, deltaX, deltaY);
         }
     }
 
     @Inject(method = "mouseReleased(DDI)Z", at = @At("RETURN"))
-    private void mouseReleasedPre(double mouseX, double mouseY, int button, CallbackInfoReturnable<Boolean> cir) {
-        TextInputBoxWidget focusedTextInput = ((TextboxScreen) this).getFocusedTextInput();
-
-        if (focusedTextInput != null) {
-            focusedTextInput.mouseReleased(mouseX, mouseY, button);
+    private void mouseReleasedPost(double mouseX, double mouseY, int button, CallbackInfoReturnable<Boolean> cir) {
+        for (GuiEventListener listener : ((AbstractContainerScreen<?>) (Object) this).children()) {
+            listener.mouseReleased(mouseX, mouseY, button);
         }
     }
```

Comment:
What's the rationale behind these changes? I think this definitely breaks *something*.

Reply:
Old version only sent mouseReleased events to the focusedTextInput, this will send it to all children

Reply:
And what does this solve for us? The original method is VERY long, I am not sure if we benefit from this change affecting all abstract container screens. If this is only a "fix" for a single feature, it might be wiser to add an event and hook to mouseReleased there.

Reply:
This solves children of AbstractContainerScreens not receiving any mouse events at all. It doesn't exactly make sense in my head to exclude children of these screens except TextInputs, I can't think of anything, but if there was a specific reason to do this, lmk

Reply:
But is this something that _needs_ to be fixed? AbstractContainerScreen were designed by the original author, not to handle mouseReleased, as it's a too special case compared to regular screens. So, what is there to fix? Does this fix a single screen? Does it fix more? If it fixes a single screen, then I think mouseReleased should be fixed only in that case.

Reply:
@ryanzhoudev It is often better to handle controversial changes separately. Is this fix *really required* for this PR? If not, leave it out and open a separate PR. If yes, create a separate PR with just this fix, and integrate it first, so we can focus on just that particular change.